src/graph.jl

```diff
@@ -148,6 +150,12 @@ function status(sf::StackFrame)
     if !sf.from_c && sf.func === :eval_user_input && endswith(String(sf.file), "REPL.jl")
         st |= repl
     end
+    if !sf.from_c && occursin("./compiler/", String(sf.file))
+        st |= compilation
+    end
+    if !sf.from_c && occursin("task.jl", String(sf.file))
```

Comment:
Should probably be 
```suggestion
    if !sf.from_c && endswith(String(sf.file), "task.jl")
```
Ideally we'd check that all children of this node are in task.jl as well, but that seems pretty expensive.

Reply:
I think the best thing would be to use the new `sleeping` metadata field, but that's not currently threaded through to here

Reply:
Perhaps the right thing to do for 1.8 is to expand Base.StackFrame to include the metadata https://github.com/JuliaLang/julia/blob/cbf6c1c8965aaba00683bfe4d54be9889368c1f6/base/stacktraces.jl#L48

Reply:
I guess the stacktrace show methods could then show some of the metadata like threadid, taskid, time?

Reply:
Could we look specifically for scheduler-related events? I'd guess that's the main expense of task-handling? (I've spent very little time investigating this myself, so it may be a naive question.)